README.md

```diff
@@ -1,216 +1,222 @@
+This repo contains contracts enabling onchain staging of contract updates, providing mechanisms to store code,
+delegate update capabilities, and execute staged updates.
 
-## Simple Case Demo
+## Overview
 
-For this run through, we'll focus on the simple case where a single contract is deployed to a single account that can
-sign the setup & delegation transactions. 
+> :information_source: This document proceeds with an emphasis on the `MigrationContractStaging` contract, which will be
+> used for the upcoming Cadence 1.0 network migration. Any contracts currently on Mainnet **WILL** need to be updated
+> via state migration on the Cadence 1.0 milestone. This means you **MUST** stage your contract updates before the
+> milestone for your contract to continue functioning. Keep reading to understand how to stage your contract update.
 
-This use case is enough to get the basic concepts involved in the `StagedContractUpdates` contract, but know that more
-advanced deployments are possible with support for multiple contract accounts and customized deployment configurations.
+The `MigrationContractStaging` contract provides a mechanism for staging contract updates onchain in preparation for
+Cadence 1.0. Once you have refactored your existing contracts to be Cadence 1.0 compatible, you will need to stage your
+code in this contract for network state migrations to take effect and your contract to be updated with the Height
+Coordinated Upgrade.
 
-### Setup
+### `MigrationContractStaging` Deployments
 
-1. Start your local emulator:
+> :information_source: The `MigrationContractStaging` contract is not yet deployed. Its deployment address will be added
+> here once it has been deployed.
 
-    ```sh
-    flow emulator
-    ```
-
-1. Setup emulator environment - this creates our emulator accounts & deploys contracts:
-
-    ```sh
-    sh setup.sh
-    ```
-
-### Walkthrough
+| Network | Address |
+| ------- | ------- |
+| Testnet | TBD     |
+| Mainnet | TBD     |
 
-1. We can see that the `Foo` has been deployed, and call its only contract method `foo()`, getting back `"foo"`:
+### Pre-Requisites
 
+- An existing contract deployed to your target network. For example, if you're staging `A` in address `0x01`, you should
+  already have a contract named `A` deployed to `0x01`.
+- A Cadence 1.0 compatible contract serving as an update to your existing contract. Extending our example, if you're
+  staging `A` in address `0x01`, you should have a contract named `A` that is Cadence 1.0 compatible. See the references
+  below for more information on Cadence 1.0 language changes.
+- Your contract as a hex string.
+  - Included in this repo is a Python util to hex-encode your contract which outputs your contract's code as a hex
+    string. With Python installed, run:
     ```sh
-    flow scripts execute ./scripts/test/foo.cdc
+    python3 ./src/get_code_hex.py <PATH_TO_YOUR_CONTRACT>
```

Comment:
Awesome, thanks! Ideally we'd integrate it all into the CLI so that contract aliases are also replaced, but for now your command is much better than using Python. I'll replace it now